File: Lab 1/Новикова А.С/src/stack.cpp
#include "stack.h"

Stack::Stack(int elemAmount)
{
	top = -1;
}
Stack::~Stack()
{
	top = 0;
}

std::string Stack::peak()
{
	if (!isEmpty())
		return data[top];

	std::string noPeak = "The stack is empty";
	return noPeak;
}

void Stack::pop()
{
	if (isEmpty())
		std::cout << "The stack is empty! Nothing to pop!";

	else
		top--;

}

void Stack::push(std::string value)
{
	if (isFull())
	{
		data.pushBack(value);
		top++;
	}

	else
		data[++top] = value;
}

bool Stack::isEmpty()
{
	return (top < 0);
}

bool Stack::isFull()
{
	return (top == data.getSize() - 1);
}

File: Lab 1/Таканаев Т.Ш/src/ru/Timur141234/Structures/DynamicStack/Stack.java
package ru.Timur141234.Structures.DynamicStack;

/**
 *
 * @author Timur Takanaev
 */
public class Stack<T> {

    private T[] array; // Массив, хранящий данные.
    private int top; // Индекс элемента, находящегося в вершине стека.
    private int capacity; // Вместимость стека.

    public Stack(int size) {

        this.array = (T[]) new Object[size];
        this.top = -1;
        this.capacity = size;

    }

    public void push(T value) {

        if (!isFull()) {

//            System.err.println("Добавляем " + value);
            top++;
            array[top] = value;

        } else {

            System.err.println("В стеке больше нет места!");
            System.exit(0);

        }

    }

    public T pop() {

        if (isEmpty()) {

            System.err.println("В стеке отсутствуют какие-либо элементы!");
            System.exit(0);

        }

        T value = array[top];

//        System.err.println("Удаляем " + value);
        array[top] = null;
        top--;

        return value;
    }

    public T peek() {

        if (isEmpty()) {

            System.err.println("В стеке отсутствуют какие-либо элементы!");
            System.exit(0);

        }

        return array[top];
    }

    public int getSize() {
        return top + 1;
    }

    public boolean isEmpty() {
        return getSize() == 0;
    }

    public boolean isFull() {
        return getSize() == capacity;
    }
}
